modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/session.es.js

```diff
@@ -41,10 +41,13 @@ function getSessionClickURL() {
  * @return {Promise}
  * @review
  */
-export function getSessionValue(key) {
+export function getSessionValue(key, useHttpSession) {
```

Comment:
I'm not totally sold on the idea of adding extra boolean parameters to these functions because ["duplication is better than inappropriate sharing"](https://medium.com/@amlcurran/clean-code-the-curse-of-a-boolean-parameter-c237a830b7a3).

Also, I'm not really sure why we add them to both — I see us `set()`-ting `useHttpSession`, but never passing it when `get()`-ting.

Reply:
maybe because sometimes we read the value directly with the java class com.liferay.portal.kernel.util.SessionClicks , but we should expose the "endpoint" to the front in this util.

Reply:
If we do go with changing the API (which I am not convinced of), doc comments will need to be updated — just leaving this comment here as a reminder to do that.

Reply:
If we are not going to change the api we continue use the old style:

Reply:
That seems reasonable to me, in the spirit of making minimal changes.

Reply:
There was actually a pretty [lively conversation in Slack](https://liferay.slack.com/archives/C03LW8NUK/p1602081892045700) about `SessionClicks` and `httpSession` in the context of `User Preferences` which actually showed this was less than obvious and the behaviour here totally unexpected and unknown...

I would ask that we don't add APIs in the form of additional boolean parameters as to avoid the Boolean trap antipattern.

This can be seen in action in `Liferay.Util.Session.set('ignoreUserLocaleOptions', true, true)`

Mabye something like:

```javascript
export function getSessionValue(key, options)
export function setSessionValue(key, value, options)

// So you would be able to do
Liferay.Util.Session.set('ignoreUserLocaleOptions', true, {
    useHttpSession: true
});
```

Reply:
I think it's important to have an API that also distinguishes between what you want to do (store the value `true` for the `ignoreUserLocaleOptions`) from how you want to do it (using an http session). 

The old API failed to do this since it exposed both params in the same way simply because of the fact that all could be sent and read in the same way. There's no way to know in the old case whether we wanted to do that or to set the value `true` to the `useHttpOption` ignoring the user locale options ;)